src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -299,7 +298,10 @@ private WatchedDirectory AddDirectoryWatchUnlocked(WatchedDirectory parent, stri
 
                 // Add a watch for the full path.  If the path is already being watched, this will return 
                 // the existing descriptor.  This works even in the case of a rename.
-                int wd = (int)SysCall(fd => Interop.libc.inotify_add_watch(fd, fullPath, (uint)_notifyFilters));
+                int wd = (int)SysCall(
+                    (fd, fullpath, runningInstance) => Interop.libc.inotify_add_watch(fd, fullpath, (uint)runningInstance._notifyFilters),
```

Comment:
I'd prefer if we used a different name for the ```fullPath``` lambda parameter.  As it shadows the outer scope's variable, it'd be very easy to accidentally change the name of the parameter without changing the usage sites, and still have everything compile but accidentally reintroduce the closure.